### PR TITLE
Ronin gil/leaderboard

### PR DESCRIPTION
**Pull request by RoninGil**
Added Passport-Slack authentication with Express.
Added a Leaderboard sample with users from DB.
Changes on redux store to see the state using the Redux DevTools.

### CODE REVIEW

File: src/components/redeemModal/reedemModal.component.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { Modal, Button } from 'react-bootstrap'
 
 export default function RedeemModal(props) {
+    console.log(props)
```
Comment (manuelrdz-raweng) [inline src/components/redeemModal/reedemModal.component.jsx:5]: Remove this console log pls

File: src/components/searchBox/searchBox.component.jsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { InputGroup, FormControl } from 'react-bootstrap';
 
 export  default function SearchBox(props) {
+    console.log(props.text)
```
Comment (manuelrdz-raweng) [inline src/components/searchBox/searchBox.component.jsx:5]: Remove this one as well

File: src/components/usersLeaderboard/usersLeaderboard.component.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            const realUsers = users.filter(user => {
+                return user.real_name
+            });
+            setUsersDB(realUsers.sort((a, b) => {
+                return (b.total_coins) - (a.total_coins)
+            }));
+            
+        }
+        getUsersDB();
+        console.log(usersDB)
```
Comment (manuelrdz-raweng) [inline src/components/usersLeaderboard/usersLeaderboard.component.jsx:20]: Remove this console log

File: src/db/index.js [hunk 4]
```diff
@@ -1,22 +1,43 @@
...
+    firebase.auth().signInWithEmailAndPassword(process.env.REACT_APP_EMAIL, process.env.REACT_APP_PASS)
+        .then(() => console.log('AUTHENTICATED'))
+        .catch((e) => console.error(`Authentiaction failed: ${e}`))
 }
 
 export const fetchRewards = async () => { 
     const snapshot = await firebase.database().ref('rewards').once('value')
     const rewardsData = snapshot.val()
-
+    console.log(Object.keys(rewardsData).map(id => ({...rewardsData[id], id})))
```
Comment (manuelrdz-raweng) [inline src/db/index.js:24]: Remove this console log

File: src/pages/home/home.page.jsx [hunk 5]
```diff
@@ -1,5 +1,22 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getUserLoggedIn, startGetUser } from '../../redux/user/user.actions';
 
 export default function Home() {
-    return <h2>Home</h2>;
+    const {currentUser} = useSelector(state => state.user);
+    const dispatch = useDispatch();
+    console.log(currentUser.uid)
```
Comment (manuelrdz-raweng) [inline src/pages/home/home.page.jsx:8]: Remove this one as well

File: src/redux/user/user.selectors.js [hunk 6]
```diff
@@ -1 +1,8 @@
-export const getCurrentUser = state => state.user.currentUser;
\ No newline at end of file
+export const getCurrentUser = state => state.user.currentUser;
+
+export const getLoggedUser = async() => {
+    const getData = await fetch('/user');
+    const getBody = await getData.json()
+    console.log(getBody.user);
```
Comment (manuelrdz-raweng) [inline src/redux/user/user.selectors.js:6]: Remove this console log pls

File: server.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+const express = require('https-localhost');
+// const jwt = require('jsonwebtoken');
+// const bodyParser = require('body-parser');
```
Comment (manuelrodgzz) [inline server.js:3]: If we are not planning to use this libraries in the future please delete this comments. Do the same with the comment on line 6 please

**Comment by manuelrodgzz** [comment]
LGTM

**Review by Fermich11 — APPROVED** [review]

**Pull request merged**